### PR TITLE
diffusion_1 pde missing time dependence?

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/project-asgard/asgard/wiki/developing)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Time dependence of the analytic solution appears to be broken on at least `diffusion_1` and `diffusion_2`. This drops relative difference (numeric-analytic) from ~15% to ~0.5%, but I'm hoping there's a better way.

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [ ] Bugfix
- [ ] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- [ ] this PR is up to date with current the current state of 'develop'
- [ ] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

**Comment by stefan-schnake** [comment]
`exact_time` should be called when creating the analytic solution that changes in time.  Is this not the case?

**Comment by quantumsteve** [comment]
> `exact_time` should be called when creating the analytic solution that changes in time. Is this not the case?

exact_time doesn't appear to be called anymore...

Here's out last tagged version
https://github.com/project-asgard/asgard/blob/v0.4.0/src/main.cpp#L190

current `develop`
https://github.com/project-asgard/asgard/blob/27f5d245d4310c3ffa0785ebb3733eb5d8578796/src/transformations.cpp#L407
https://github.com/project-asgard/asgard/blob/27f5d245d4310c3ffa0785ebb3733eb5d8578796/src/transformations.cpp#L425

File: src/pde/pde_base.hpp [hunk 1]
```diff
@@ -549,7 +549,7 @@ class PDE
       // each set of analytical solution functions must have num_dim functions
       for (const auto &md_func : exact_vector_funcs)
       {
-        expect(md_func.size() == static_cast<unsigned>(num_dims));
+        expect(std::clamp(md_func.size(), static_cast<size_t>(num_dims), static_cast<size_t>(num_dims) + 1u) == md_func.size());
```
Comment (mkstoyanov) [inline src/pde/pde_base.hpp]: Isn't this just:
```
md_func.size() == static_cast<size_t>(num_dims) or md_func.size() == static_cast<size_t>(num_dims + 1)
```
Clamping the func-size just to compare to the func-size is weird.

File: src/transformations.hpp [hunk 2]
```diff
@@ -242,7 +242,7 @@ inline void transform_and_combine_dimensions(
     P const time_multiplier, fk::vector<P, mem_type::view> result)
 
 {
-  expect(v_functions.size() == dims.size());
+  expect(std::clamp(v_functions.size(), dims.size(), dims.size() + 1u) == v_functions.size());
```
Comment (mkstoyanov) [inline src/transformations.hpp]: Same clamp trick, the dim-size and dim-size+1 are only one unit apart.

**Review by mkstoyanov — APPROVED** [review]

**Pull request merged**